pkg/logger/ecs_error_log.go

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package logger
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"go.uber.org/zap/zapcore"
+)
+
+// ECSErrorLogger is used to send the error level log to a separated file and rotate
+// error logs when expired, ECS agent will mount the filepath and read it periodically
+type ECSErrorLogger struct {
+	ErrorLogFilePath    string
+	ErrorLogMaxAge      int
+	ErrorLogFileMaxSize int
+
+	errorChannel chan zapcore.Entry
+	queue        []zapcore.Entry
+	ticker       time.Ticker
+	errorFile    *os.File
+	size         int
+}
+
+// NewECSErrorLogger could new a struct based on the 3 environment variables:
+// error file path, error log TTL and error file max size
+func NewECSErrorLogger() *ECSErrorLogger {
+	ecsErrorLogFilePath := os.Getenv("STATUS_MESSAGE_FILE_PATH")
+	errorLogMaxAge, err := strconv.Atoi(os.Getenv("STATUS_MESSAGE_TTL"))
+	if err != nil || errorLogMaxAge <= 0 {
+		log.Fatalf("failed to get STATUS_MESSAGE_TTL: %v", err)
```

Comment:
I don't think that failure to set up this logging handler should be fatal to the application.  Have it return an error that can be logged but don't use `log.Fatal`.

Reply:
since this is a requirement for the error reporting feature, if user choose to use it, they have to provide a `STATUS_MESSAGE_TTL`

Reply:
Or there can be a reasonable default.  This is should not be fatal to the process.

Reply:
we had a talk before about it, if customer choose to use this feature and not provide a ttl parameter, we will exit the process

Reply:
Who is "we"? Who from ADOT approved that decision?  This is contrary to the design principles followed by most OTel components and may be surprising to users.  I don't think it is the correct decision when a default value can be provided.  How is the user to know what value to use for this TTL?  If it is something that the control plane is supposed to inject automatically then we can set a default here that mirrors what should be injected.  If it is something that we provide recommendations on through documentation then we can set the default to match that recommendation.

Reply:
I did the high level design for this... and while the backend will always set these values, it doesn't hurt to have a default.